File: lab3/src/main/java/lab3/functionality/Action.java
package lab3.functionality;

public class Action extends Nameable {
    private final boolean disposable;
    private final Usable usable;

    public Action(String name, Description description, Usable usable) {
        this(name, description, usable, false);
    }

    public Action(String name, Description description, Usable usable, boolean disposable) {
        super(name, description);
        this.usable = usable;
        this.disposable = disposable;
    }

    public void use(Player player) {
        usable.use(player);
    }

    public boolean isDisposable() {
        return disposable;
    }
}

File: lab4/src/main/kotlin/lab4/functionality/Action.kt
package lab4.functionality

class Action(
    name: String,
    description: Description,
    private val usable: Usable,
    val isDisposable: Boolean = false
) : Nameable(name, description) {

    fun use(player: Player, obj: Actionable) = usable.use(player, obj)
}